apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/resources/bfd-manifest.xsd

```diff
@@ -12,6 +12,22 @@
             <xs:attribute name="exportType" type="xs:string" use="optional" />
           </xs:complexType>
         </xs:element>
+        <xs:element name="syntheaEndStateProperties" minOccurs="0" maxOccurs="1">
```

Comment:
I thought we were in agreement that this interface should not be coupled to synthea or end-state properties (a synthea concept) and instead be based around the more general concept of pre-validation of a load. Having the pipeline perform optional pre-validation is a better separation of concerns than having it be coupled to synthea concepts which could change. The coupling to synthea concepts should only be present in the jenkins job which has knowledge of synthea and the pipeline.

Reply:
Well sort of; so I can certainly change the name of the element (remove any trace of synthea), as well as remove any biding between the isSynthetic attribute and the checking of the pre-validation elements. However, it is VERY unlikely that upstream sources of a manifest have the same level of constraints or 'pre-knowledge' of meta-data to imbue into a manifest.

Reply:
The notion of synthetic data is not synthea specific so I don't see an issue there. This is less about leaving the possibility of another upstream open (although, who knows for sure?) and more about not tightly coupling all of the various components together when they don't need to be.

Reply:
Taking this conversation a bit further, the individual elements I believe could also use some renaming to support a more intuitive _pipeline_ interface for validation. For instance, clm_grp_id_start in this context doesn't say where the clm_grp_ids for _this_ load start, it says where the clm_grp_ids for the next synthea generated data should start. A more intuitive validation interface would describe this load itself so I suggest naming this clm_grp_id_max or clm_grp_id_end and passing the value from the end state properties minus 1. This keeps the knowledge of synthea as much out of the pipeline as possible.

Reply:
Agreed (mostly); where things get a little hairy when trying to completely synthea-agnostic, is the following:
- the db function that is called performs range checking, much of which is specific to Synthea data; for example a range check of bene_id (_p_beg_bene_id, p_end_bene_id_) results in a where clause _bene_id <= p_beg_bene_id AND bene_id > p_end_bene_id_. This however is ONLY valid for synthetic data which use negative bene_id values; if fact for that same range check for non-synthetic data it would be: _bene_id > p_beg_bene_id AND bene_id <= p_end_bene_id_.
- ideally, for any validation that involves range checking, both the _beg_<whatever column> and _end_<whatever column> shod be passed in.

So I guess one could derive the LT vs. GT value checking by checking the sign of the passed in variable (i.e., p_beg_bene_id):
- p_beg_bene_id < 0 : where bene_id <= p_beg_bene_id
- p_beg_bene_id > 0 : where bene_id > p_beg_bene_id

This is starting to feel like significant AC creep; I'd prefer get this in under a mostly Synthea required functionality and a subsequent ticket to make more agnostic.

Reply:
My main concern is the interface -- the manifest XML naming and I think that should be part of this so that the dependent work can be completed and not have to be changed later. The internal method naming is secondary and could be changed in a later ticket if that is burdensome. I would not expect any additional validation support beyond what is needed for this use case but the things that are needed for this use case should be well named in the interface (manifest).

Reply:
So the interface as now a full-fledged Java interface that leaves the validation up to the implementation class the implements the interface. The only mention of the word 'synthea' is when instantiating the implementation that is associated with Synthea data. All pre-validation manifest content has been changed to be more inclusive to non-synthea interfaces.